### PR TITLE
Add RBAC v1beta1

### PR DESCRIPTION
**Pull request by deads2k**
Add `rbac.authorization.k8s.io/v1beta1`.  This scrubs `v1alpha1` to remove cruft, then add `v1beta1`.  We'll update other bits of infrastructure to code to `v1beta1` as a separate step.

```release-note
The `attributeRestrictions` field has been removed from the PolicyRule type in the rbac.authorization.k8s.io/v1alpha1 API. The field was not used by the RBAC authorizer.
```

@kubernetes/sig-auth-misc @liggitt @erictune

### CODE REVIEW

File: pkg/generated/bindata.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func (fi bindataFileInfo) Sys() interface{} {
...
...96\xf0\x89\xfa\x8b\x5b\xb2\x67\xbc\x70\xad\x55\x9b\x77\x9f\xe2\x1e\xba\x80\xa9\x77\x1f\xb9\x8a\x34\x52\xf4\x9e\x7d\xd0\xce\x16\x58\x73\x8c\x7c\x15\xa9\x76\xc4\x57\x72\xd1\x18\x0e\x34\x67\x63\x5c\x72\xca\x5d\x05\x34\x09\xb5\x56\x74\xd0\xd6\x81\x4a\x57\x60\x32\x3d\x29\xe9\xd0\xf3\xea\x4f\xd0\x1b\x19\xb9\xc0\x5c\x64\xdb\x94\xe5\x94\xe5\x98\x8b\x42\x6c\x3f\x13\x42\x88\x64\x7a\x42\xa7\x7c\xa9\xc3\x3f\xdc\xee\x8a\xdb\xc6\x3c\x2b\x5e\xec\x90\xd8\x13\x22\x19\xcb\x10\xa9\xf4\xd2\x06\x23\xa3\xf3\x05\x1e\x78\xb6\x4a\x5a\x3c\x68\xbd\x0d\xf8\x72\xd6\xcb\x54\xa5\xb3\x2e\xf0\xba\x5e\x48\x6d\xd2\xca\x2d\x5e\x25\x93\xd1\xe4\xe8\xa6\x95\x2c\x15\xc9\xa1\xb3\x91\x6d\xa4\xf2\xba\xe1\x02\xbb\xce\x9e\x37\x46\x6a\xbb\x8f\xd5\x5c\xfa\xc0\x71\x78\x56\x1e\xd3\xde\x0d\xd7\xe5\xbd\x60\x4f\x47\xb6\x72\x4a\xdb\xba\xc0\xbd\x99\x8e\xc9\x39\xbd\x65\xcb\xbe\x2f\x68\xea\x58\xe9\x88\x59\xba\x9b\x66\x22\x39\xa7\x5e\xd3\xbb\xd5\x84\x0f\x7b\xdf\x02\x7b\xe3\xb1\xb4\x75\x2b\x6b\xa6\x92\xe5\xa2\x1b\x97\x69\xbd\x34\x74\xec\xfc\x22\x14\x68\x9b\x95\x0c\xc3\x7c\x1f\xfb\xd7\xe1\x96\xc5\x47\x43\xcc\x9e\xee\xaf\x3f\x2d\x90\x6d\x72\xab\xfb\x71\x47\x4b\x83\x4b\x19\xee\x6f\x2b\x7f\x05\x00\x00\xff\xff\xaa\xa3\x15\xc3\x6a\x04\x00\x00")
```
Comment (liggitt) [inline pkg/generated/bindata.go]: spurious diff

File: pkg/apis/rbac/v1beta1/defaults.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
Comment (liggitt) [inline pkg/apis/rbac/v1beta1/defaults.go]: 2017 on new files

File: pkg/apis/rbac/v1beta1/types.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	APIGroups []string `json:"apiGroups,omitempty" protobuf:"bytes,2,rep,name=apiGroups"`
+	// Resources is a list of resources this rule applies to.  ResourceAll represents all resources.
+	// +optional
+	Resources []string `json:"resources,omitempty" protobuf:"bytes,3,rep,name=resources"`
+	// ResourceNames is an optional white list of names that the rule applies to.  An empty set means that everything is allowed.
+	// +optional
+	ResourceNames []string `json:"resourceNames,omitempty" protobuf:"bytes,4,rep,name=resourceNames"`
+
+	// NonResourceURLs is a set of partial urls that a user should have access to.  *s are allowed, but only as the full, final step in the path
+	// This name is intentionally different than the internal type so that the DefaultConvert works nicely and because the ordering may be different.
```
Comment (liggitt) [inline pkg/apis/rbac/v1beta1/types.go]: remove the "intentionally different" comment... doesn't apply in the k8s port

**Comment by deads2k** [comment]
Other comments?  We're starting to see issues come from other controllers, so it looks like the enforcement is working.

**Comment by deads2k** [comment]
@cjcullen Based on prior history, I'm betting this is failing a test in GKE that I can't see.  It removes a dead field from an alpha API (see first commit).

**Comment by liggitt** [comment]
> It removes a dead field from an alpha API (see first commit).

did you remove the dead field from all yaml files in tree? client-side validation could be failing loading policy

**Comment by deads2k** [comment]
> did you remove the dead field from all yaml files in tree? client-side validation could be failing loading policy

We never put it in any.

**Comment by cjcullen** [comment]
GKE won't pass unless you either undo the removal of ABAC, or transition all of the bindings in cluster/addons to be v1beta1.

**Comment by cjcullen** [comment]
But then I'm not sure why GCE works...

**Comment by deads2k** [comment]
rebased.  @liggitt @gmarek this creates the API version its trying to start.

File: pkg/apis/rbac/v1alpha1/types.go [hunk 4]
```diff
@@ -48,10 +47,6 @@ const (
 type PolicyRule struct {
 	// Verbs is a list of Verbs that apply to ALL the ResourceKinds and AttributeRestrictions contained in this rule.  VerbAll represents all kinds.
 	Verbs []string `json:"verbs" protobuf:"bytes,1,rep,name=verbs"`
-	// AttributeRestrictions will vary depending on what the Authorizer/AuthorizationAttributeBuilder pair supports.
```
Comment (liggitt) [inline pkg/apis/rbac/v1alpha1/types.go]: as long as v1alpha1 is around, need to tombstone this protobuf index so we don't reuse it

Comment (deads2k) [inline pkg/apis/rbac/v1alpha1/types.go]: > as long as v1alpha1 is around, need to tombstone this protobuf index so we don't reuse it

Current numbers skip it.  We should be  able to do something more formal (comment I guess) after this fire is out.

**Comment by liggitt** [comment]
nit on protobuf tombstone, LGTM otherwise

**Comment by liggitt** [comment]
needs release note for alpha field removal

**Comment by gmarek** [comment]
Waiting for green before merging.

**Comment by deads2k** [comment]
Looks real :(

**Comment by deads2k** [comment]
It's a one liner contained to the integration test.  I want to wait for the other tests to finish before I push.  They were run on fb27fe0a1b72b07e4e1a9b15e85b1f4092e8e392

**Comment by deads2k** [comment]
All other required jobs are green.  Failure is here https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/39625/pull-kubernetes-unit/12616/, pushing one more commit that gets this

```
[deads@deads-dev-01 kubernetes]$ hack/test-integration.sh auth
NOTE: hack/test-integration.sh has been replaced by 'make test-integration'

The equivalent of this invocation is: 
    make test-integration WHAT='auth'


+++ [0111 09:25:58] Checking etcd is on PATH
/home/deads/workspaces/kube/src/k8s.io/kubernetes/third_party/etcd/etcd
+++ [0111 09:25:58] Starting etcd instance
etcd --advertise-client-urls http://127.0.0.1:2379 --data-dir /tmp/tmp.PLUStw32Sb --listen-client-urls http://127.0.0.1:2379 --debug > "/dev/null" 2>/dev/null
Waiting for etcd to come up.
+++ [0111 09:25:59] On try 2, etcd: : http://127.0.0.1:2379
{"action":"set","node":{"key":"/_test","value":"","modifiedIndex":4,"createdIndex":4}}
+++ [0111 09:25:59] Running integration test cases
Running tests for APIVersion: v1,apps/v1beta1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1beta1,autoscaling/v1,batch/v1,batch/v2alpha1,certificates.k8s.io/v1alpha1,extensions/v1beta1,imagepolicy.k8s.io/v1alpha1,policy/v1beta1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,storage.k8s.io/v1beta1
+++ [0111 09:26:01] Running tests without code coverage
ok  	k8s.io/kubernetes/test/integration/auth	20.795s
+++ [0111 09:26:27] Cleaning up etcd
+++ [0111 09:26:27] Integration test cleanup complete
+++ [0111 09:26:27] Cleaning up etcd
+++ [0111 09:26:27] Integration test cleanup complete
```

**Comment by liggitt** [comment]
squash the test fix when rebasing

**Comment by deads2k** [comment]
This requires a coordinated change with CI, so I'm bumping priority in advance.

**Comment by deads2k** [comment]
@k8s-bot bazel test this

**Comment by deads2k** [comment]
@cjcullen and hour to finish tests once and go into the queue and an hour to merge, I'm tagging this now.

**Comment by deads2k** [comment]
@cjcullen looks like its right on schedule.

**Pull request merged**